lighthouse-core/config/budget.js

```diff
@@ -240,6 +240,38 @@ class Budget {
     };
   }
 
+  /**
+   * @param {string} origin
+   * @return {string}
+   */
+  static validateOrigin(origin) {
+    if (origin.endsWith(`/`)) {
+      throw new Error(`Invalid origin: ${origin}. Origin should not contain a trailing '/'.`);
+    }
+    let verifiedOrigin;
+    try {
+      verifiedOrigin = new URL(origin).origin;
+    } catch (err) {
+      throw new Error(`${origin} is not a valid URL.`);
+    }
+    if (origin !== verifiedOrigin) {
+      throw new Error(`Invalid origin: ${origin}. Did you mean ${verifiedOrigin}?`);
```

Comment:
this feels a tad rigid and might make it kindof a pain to opt in an entire root domain, maybe there's a doc I've forgotten that discusses some of the alternatives considered? :)